quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```diff
@@ -52,36 +61,36 @@ public Collection findCollection(String id) throws InvalidObjectIdException {
         }
     }
 
-    /**
-     * 
-     * Edits an existing Collection and updates it in the db.
-     * 
-     * @param id of the collection that needs to be updated
-     * @param name will be the updated name value
-     * @param description will be the updated description value
-     * @return Collection Instance that is updated in database
-     * @throws CollectionNotFoundException in case the collection for the given id is missing
-     * @throws InvalidObjectIdException if collectionId couldn't be converted to ObjectId
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.ICollectionManager#editCollection(java.lang.String, java.lang.String, java.lang.String, java.util.List)
      */
     @Override
-    public Collection editCollection(String id, String name, String description) throws CollectionNotFoundException, InvalidObjectIdException {
+    public Collection editCollection(String id, String name, String description, List<String> apps) throws CollectionNotFoundException, CitesphereAppNotFoundException, InvalidObjectIdException {
         Collection collection = findCollection(id);
 
         if (Objects.nonNull(collection)) {
+            List<CitesphereAppInfo> citesphereApps = citesphereConnector.getCitesphereApps();
+            HashSet<String> appSet = new HashSet<String>(
+                    citesphereApps.stream().map(app -> app.getClientId()).collect(Collectors.toList()));
+            
+            if (apps != null) {
+                for (String clientId : apps) {
+                    if (!appSet.contains(clientId)) {
+                        throw new CitesphereAppNotFoundException("No Citesphere App found with client id " + clientId);
+                    }
+                }
+            }
```

Comment:
this looks like duplciate code (same as in addcollection) and should be more DRY